qcfractal/tests/test_collections.py::test_torsiondrive_dataset is flaky on Travis

**Describe the bug**
The test_torsiondrive_dataset test fails/passes at random on Travis in the *.4 pipeline.

**To Reproduce**
Run the *.4 pipeline on master repeatedly until the bug occurs.

**Expected behavior**
All pipelines should always pass on master.

**Additional context**
Full error output below. 

```
__________________________ test_torsiondrive_dataset ___________________________

fractal_compute_server = FractalServer(name='QCFractal Snowflake Instance' uri='http://localhost:59745/')

    @pytest.mark.slow

    @testing.using_torsiondrive

    @testing.using_geometric

    @testing.using_rdkit

    def test_torsiondrive_dataset(fractal_compute_server):

    

        client = ptl.FractalClient(fractal_compute_server)

    

        ds = ptl.collections.TorsionDriveDataset("testing", client=client)

    

        hooh1 = ptl.data.get_molecule("hooh.json")

        hooh2 = hooh1.copy(update={"geometry": hooh1.geometry + np.array([0, 0, 0.2])})

    

        ds.add_entry("hooh1", [hooh1], [[0, 1, 2, 3]], [90], attributes={"something": "hooh1"})

        ds.add_entry("hooh2", [hooh2], [[0, 1, 2, 3]], [90], attributes={"something": "hooh2"})

    

        optimization_spec = {

            "program": "geometric",

            "keywords": {

                "coordsys": "tric",

            }

        }

        qc_spec = {

            "driver": "gradient",

            "method": "UFF",

            "basis": "",

            "keywords": None,

            "program": "rdkit",

        }

    

        ds.add_specification("Spec1", optimization_spec, qc_spec, description="This is a really cool spec")

    

        ncompute = ds.compute("spec1")

        assert ncompute == 2

        assert ds.status("spec1")["Spec1"].sum() == 2  # Might have completed from previous run.

    

        ds.save()

    

        fractal_compute_server.await_services(max_iter=1)

    

        # Check status

        status_detail = ds.status("Spec1", detail=True)

        assert status_detail.loc["hooh2", "Complete"] == 1

        assert status_detail.loc["hooh2", "Total Points"] == 4

    

        fractal_compute_server.await_services(max_iter=5)

    

        ds = client.get_collection("torsiondrivedataset", "testing")

        ds.query("spec1")

    

        # Add another fake set, should instantly return

        ds.add_specification("spec2", optimization_spec, qc_spec, description="This is a really cool spec")

    

        # Test subsets

        ncompute = ds.compute("spec2", subset=set())

        assert ncompute == 0

    

        ncompute = ds.compute("spec2")

        assert ncompute == 2

        ds.query("spec2")

    

        # We effectively computed the same thing twice with two duplicate specs

        for row in ["hooh1", "hooh2"]:

            for spec in ["Spec1", "spec2"]:

>               assert pytest.approx(ds.df.loc[row, spec].get_final_energies(90), 1.e-5) == 0.00015655375994799847

qcfractal/tests/test_collections.py:900: 

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

qcfractal/interface/models/torsiondrive.py:229: in get_final_energies

    return self._organize_return(self.final_energy_dict, key)

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <TorsionDriveRecord(id='12' status='INCOMPLETE')>, data = {}

key = '[90]', minimum = False

    def _organize_return(self, data: Dict[str, Any], key: Union[int, str, None],

                         minimum: bool = False) -> Dict[str, Any]:

    

        if key is None:

            return {self._deserialize_key(k): copy.deepcopy(v) for k, v in data.items()}

    

        key = self._serialize_key(key)

    

        if minimum:

            minpos = self.minimum_positions[self._serialize_key(key)]

            return copy.deepcopy(data[key][minpos])

        else:

>           return copy.deepcopy(data[key])

E           KeyError: '[90]'

qcfractal/interface/models/torsiondrive.py:163: KeyError

----------------------------- Captured stdout call -----------------------------

1 initial tasks pushed to opt_queue

 + - - -

1 initial tasks pushed to opt_queue

 + - - -

Warning! dihedral values inconsistent with target grid_id

dihedral_values [-125.25216992801097] ref_grid_id (-90,)

1 initial tasks pushed to opt_queue

 + - - -

1 initial tasks pushed to opt_queue

 + - - -

First energy for grid_id (-90,) = 0.00015655516263484652

 o + - +

Warning! dihedral values inconsistent with target grid_id

dihedral_values [-125.25216992801097] ref_grid_id (-90,)

Warning! dihedral values inconsistent with target grid_id

dihedral_values [-125.25216992801097] ref_grid_id (-90,)

1 initial tasks pushed to opt_queue

 + - - -

1 initial tasks pushed to opt_queue

 + - - -

First energy for grid_id (-90,) = 0.00015655516263484652

 o + - +

First energy for grid_id (0,) = 0.0025975414985495403

First energy for grid_id (180,) = 0.0007534925573592105

 + o + o

Warning! dihedral values inconsistent with target grid_id

dihedral_values [-125.25216992801097] ref_grid_id (-90,)

Warning! dihedral values inconsistent with target grid_id

dihedral_values [-125.25216992801097] ref_grid_id (-90,)

Warning! dihedral values inconsistent with target grid_id

dihedral_values [-125.25216992801097] ref_grid_id (-90,)

1 initial tasks pushed to opt_queue

 + - - -

1 initial tasks pushed to opt_queue

 + - - -

First energy for grid_id (-90,) = 0.00015655516263484652

 o + - +

First energy for grid_id (0,) = 0.0025975414985495403

First energy for grid_id (180,) = 0.0007534925573592105

 + o + o

First energy for grid_id (90,) = 0.00015655376184288845

 o + o +

Warning! dihedral values inconsistent with target grid_id

dihedral_values [-125.25216992801097] ref_grid_id (-90,)

Warning! dihedral values inconsistent with target grid_id

dihedral_values [-125.25216992801097] ref_grid_id (-90,)

Warning! dihedral values inconsistent with target grid_id

dihedral_values [-125.25216992801097] ref_grid_id (-90,)

Warning! dihedral values inconsistent with target grid_id

dihedral_values [-125.25216992801097] ref_grid_id (-90,)

1 initial tasks pushed to opt_queue

 + - - -

1 initial tasks pushed to opt_queue

 + - - -

First energy for grid_id (-90,) = 0.00015655516263484652

 o + - +

First energy for grid_id (0,) = 0.0025975414985495403

First energy for grid_id (180,) = 0.0007534925573592105

 + o + o

First energy for grid_id (90,) = 0.00015655376184288845

 o + o +

All optimizations converged at lowest energy. Job Finished!

Warning! dihedral values inconsistent with target grid_id

dihedral_values [-125.25216992801097] ref_grid_id (-90,)

Warning! dihedral values inconsistent with target grid_id

dihedral_values [-125.25216992801097] ref_grid_id (-90,)

Warning! dihedral values inconsistent with target grid_id

dihedral_values [-125.25216992801097] ref_grid_id (-90,)

Warning! dihedral values inconsistent with target grid_id

dihedral_values [-125.25216992801097] ref_grid_id (-90,)

Warning! dihedral values inconsistent with target grid_id

dihedral_values [-125.25216992801097] ref_grid_id (-90,)

1 initial tasks pushed to opt_queue

 + - - -
```

--- a/qcfractal/tests/test_collections.py
+++ b/qcfractal/tests/test_collections.py
@@ -1213,11 +1213,11 @@ def test_torsiondrive_dataset(fractal_compute_server):
     hooh1 = ptl.data.get_molecule("hooh.json")
     hooh2 = hooh1.copy(update={"geometry": hooh1.geometry + np.array([0, 0, 0.2])})
 
-    ds.add_entry("hooh1", [hooh1], [[0, 1, 2, 3]], [90], attributes={"something": "hooh1"})
-    ds.add_entry("hooh2", [hooh2], [[0, 1, 2, 3]], [90], attributes={"something": "hooh2"})
+    ds.add_entry("hooh1", [hooh1], [[0, 1, 2, 3]], [60], attributes={"something": "hooh1"})
+    ds.add_entry("hooh2", [hooh2], [[0, 1, 2, 3]], [60], attributes={"something": "hooh2"})
 
     optimization_spec = {"program": "geometric", "keywords": {"coordsys": "tric"}}
-    qc_spec = {"driver": "gradient", "method": "UFF", "basis": "", "keywords": None, "program": "rdkit"}
+    qc_spec = {"driver": "gradient", "method": "UFF", "basis": None, "keywords": None, "program": "rdkit"}
 
     ds.add_specification("Spec1", optimization_spec, qc_spec, description="This is a really cool spec")
 
@@ -1232,9 +1232,9 @@ def test_torsiondrive_dataset(fractal_compute_server):
     # Check status
     status_detail = ds.status("Spec1", detail=True)
     assert status_detail.loc["hooh2", "Complete Tasks"] == 1
-    assert status_detail.loc["hooh2", "Total Points"] == 4
+    assert status_detail.loc["hooh2", "Total Points"] == 6
 
-    fractal_compute_server.await_services(max_iter=5)
+    fractal_compute_server.await_services(max_iter=7)
 
     ds = client.get_collection("torsiondrivedataset", "testing")
     ds.query("spec1")
@@ -1253,7 +1253,7 @@ def test_torsiondrive_dataset(fractal_compute_server):
     # We effectively computed the same thing twice with two duplicate specs
     for row in ["hooh1", "hooh2"]:
         for spec in ["Spec1", "spec2"]:
-            assert pytest.approx(ds.df.loc[row, spec].get_final_energies(90), 1.0e-5) == 0.00015655375994799847
+            assert pytest.approx(ds.df.loc[row, spec].get_final_energies(60), 1.0e-5) == 0.0007991272305133664
 
     assert ds.status().loc["COMPLETE", "Spec1"] == 2
     assert ds.status(collapse=False).loc["hooh1", "Spec1"] == "COMPLETE"

